wgengine/magicsock/legacy.go

```diff
@@ -480,29 +480,24 @@ func (as *addrSet) populatePeerStatus(ps *ipnstate.PeerStatus) {
 	}
 }
 
-func (a *addrSet) Addrs() []wgcfg.Endpoint {
-	var eps []wgcfg.Endpoint
+func (a *addrSet) Addrs() string {
+	var addrs []string
 	for _, addr := range a.addrs {
-		eps = append(eps, wgcfg.Endpoint{
-			Host: addr.IP.String(),
-			Port: uint16(addr.Port),
-		})
+		addrs = append(addrs, addr.String())
 	}
 
 	a.mu.Lock()
 	defer a.mu.Unlock()
 	if a.roamAddr != nil {
-		eps = append(eps, wgcfg.Endpoint{
-			Host: a.roamAddr.IP.String(),
-			Port: uint16(a.roamAddr.Port),
-		})
+		addrs = append(addrs, a.roamAddr.String())
 	}
-	return eps
+	return strings.Join(addrs, ",")
 }
 
 // singleEndpoint is a wireguard-go/conn.Endpoint used for "roaming
 // addressed" in releases of Tailscale that predate discovery
 // messages. New peers use discoEndpoint.
+// TODO(josharian): change to netaddr.IPPort.
```

Comment:
@danderson has a CL that deletes this imminently, so not worth TODOing it. Just a merge conflict for one of you.